Include/TicTacToe/Library/ApplicationController.hpp

```diff
@@ -0,0 +1,39 @@
+#ifndef TIC_TAC_TOE_APPLICATION_CONTROLLER_HPP
+#define TIC_TAC_TOE_APPLICATION_CONTROLLER_HPP
+#include <memory>
+#include <QWidget>
+#include "TicTacToe/Library/Board.hpp"
+#include "TicTacToe/Library/GameBoardWindow/GameBoardController.hpp"
+#include "TicTacToe/Library/TicTacToe.hpp"
+
+namespace TicTacToe {
+
+  /** The controller of the application. */
+  class ApplicationController {
+    public:
+
+      /**
+       * Constructs an application controller.
+       */
+      ApplicationController();
+
+      /**
+       * Release resources of the application controller.
+       */
+      ~ApplicationController();
+
+    private:
+      int m_x_badge_count;
+      int m_o_badge_count;
+      QWidget m_application_window;
+      std::unique_ptr<GameBoardController> m_game_board_controller;
```

Comment:
Can you tell me more advantages of preferring an std::optional over an std::unique_ptr except that it can reduce dynamic memory allocation? I know that std::optional usually is used as a return value.

Reply:
Doesn't have many big advantages, just a matter of consistency with our coding style.